src/index.ts

```diff
@@ -1,2 +1,17 @@
-export * from './function';
-export * from './api-gateway';
+import type FunctionContext from "./context";
+export * from "./api-gateway";
+export * from "./http";
+export * from "./triggers";
+
+type FunctionSyncHandler<Event, Result> = (
+    event: Event,
+    context: FunctionContext
+) => Result;
+type FunctionAsyncHandler<Event, Result> = (
+    event: Event,
+    context: FunctionContext
+) => Promise<Result>;
+
+export type FunctionHandler<Event, Result = unknown> =
```

Comment:
Let's set defaults for these two generics? `<Event = Http.Event, Result = Http.Result>`.

Reply:
Might be a bit misleading if it's implicit when user creates it for using in a trigger context. Moreover it might make sense to have explicitly named specific implementations of the generics like this:
```typescript
type MessageQueueHandler = FunctionHandler<MessageQueue>
// ...
```

Reply:
> Moreover it might make sense to have explicitly named specific implementations of the generics like this:

It would be awesome!
My main idea is to simplify usage of this types for most popular lambda function (which is simple `Http` one).